float64 distance
int32 control
---
Coordinates pos
